helix-term/src/ui/picker.rs

```diff
@@ -67,6 +78,161 @@ impl<T> Picker<T> {
         picker
     }
 
+    // TODO: Copied from EditorView::render_buffer, reuse
+    #[allow(clippy::too_many_arguments)]
+    fn render_buffer(
```

Comment:
This duplicates a fair bit of rendering code that is used in ui/editor. We should probably have a method that can take a Rect viewport and render it out.

Reply:
I split it into rendering the highlights and rendering the gutter in ui/editor and made them associated functions.